File: tzapi4android/src/main/java/com/tianzunchina/android/api/login/SignInUser.java
package com.tianzunchina.android.api.login;


import com.tianzunchina.android.api.utils.LoginConfig;

import java.io.Serializable;

/**
 * 登录信息实体类
 * CraetTime 2016-3-3
 * @author SunLiang
 */
public class SignInUser implements Serializable{
    public static SignInUser GUEST = new SignInUser("访客");
    String account;
    String password;
    boolean isKeep = false;
    boolean isAuto = false;

    public SignInUser(String userName, String userPassword) {
        this.account = userName;
        this.password = userPassword;
    }

    public SignInUser(String userName, String userPassword, Boolean isKeep ) {
        this.account = userName;
        this.password = userPassword;
        this.isKeep = isKeep;
    }

    public SignInUser(String userName , String userPassword ,Boolean isKeep ,
               Boolean isAuto ) {
        this.account = userName;
        this.password = userPassword;
        this.isKeep = isKeep;
        this.isAuto = isAuto;
    }

    public SignInUser(String userName) {
        this.account = userName;
    }

    public void  save(){
        LoginConfig config = new LoginConfig();
        config.saveUser(this);
    }

    public boolean noGuest(){
        return !this.equals(GUEST);
    }

    public String getAccount() {
        return account;
    }

    public String getPassword() {
        return password;
    }

    public boolean isKeep() {
        return isKeep;
    }

    public boolean isAuto() {
        return isAuto;
    }

    public String toString(){
        return "SignInUser [account="+ account +", " +
                "password="+ password +", " +
                "isAuto="+ isAuto +"]";
    }
}

File: tzapi4android/src/main/java/com/tianzunchina/android/api/login/SignInUser.kt
package com.tianzunchina.android.api.login

import com.tianzunchina.android.api.utils.LoginConfig

import java.util.HashMap

/**
 * 登录信息实体类
 * CraetTime 2016-3-3
 * @author SunLiang
 */
class SignInUser {

    var account: String
    var password: String = ""
    var isKeep = false
    var isAuto = false

    constructor(userName: String, userPassword: String) {
        this.account = userName
        this.password = userPassword
    }

    constructor(userName: String, userPassword: String, isKeep: Boolean) {
        this.account = userName
        this.password = userPassword
        this.isKeep = isKeep
    }

    constructor(userName: String, userPassword: String, isKeep: Boolean,
                isAuto: Boolean) {
        this.account = userName
        this.password = userPassword
        this.isKeep = isKeep
        this.isAuto = isAuto
    }

    constructor(userName: String) {
        this.account = userName
    }

    val data: Map<String, String> get() {
        val map = HashMap<String, String>()
        map.put(LoginConfig.USER_NAME, account!!)
        map.put(LoginConfig.USER_PASSWORD, password!!)
        return map
    }

    fun save(){
        LoginConfig.saveUser(this)
    }

    override fun toString(): String {
        return "SignInUser [account=$account, password=$password, isAuto=$isAuto]"
    }
}
